homeassistant/components/flux_led/config_flow.py

```diff
@@ -0,0 +1,231 @@
+
+from .const import (
+    CONF_AUTOMATIC_ADD,
+    CONF_CONFIGURE_DEVICE,
+    CONF_DEVICES,
+    CONF_EFFECT_SPEED,
+    CONF_REMOVE_DEVICE,
+    DEFAULT_EFFECT_SPEED,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for FluxLED/MagicHome Integration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: config_entries.ConfigEntry):
+        """Get the options flow for the Flux LED component."""
+        return OptionsFlow(config_entry)
+
+    async def async_step_import(self, data: dict = None):
+        """Handle configuration via YAML import."""
+        _LOGGER.debug("Importing configuration from YAML for flux_led")
+        config_entry = self.hass.config_entries.async_entries(DOMAIN)
+
+        if config_entry:
+            _LOGGER.warning(
+                "Your flux_led configuration has already been imported. Please remove configuration from your configuration.yaml"
+            )
+            return self.async_abort(reason="single_instance_allowed")
+
+        _LOGGER.warning(
+            "Imported auto_add configuration for flux_led. Please remove from your configuration.yaml"
+        )
+        return await self.async_step_user(
+            user_input={
+                CONF_AUTOMATIC_ADD: data[CONF_AUTOMATIC_ADD],
+                CONF_DEVICES: data.get(CONF_DEVICES, {}),
+            }
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        config_entry = self.hass.config_entries.async_entries(DOMAIN)
+        if config_entry:
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is not None:
+            devices = user_input.get(CONF_DEVICES, {})
+
+            if user_input[CONF_AUTOMATIC_ADD]:
+                scanner = BulbScanner()
+                await self.hass.async_add_executor_job(scanner.scan)
+
+                for bulb in scanner.getBulbInfo():
+                    device_id = bulb["ipaddr"].replace(".", "_")
+                    if not devices.get(device_id, False):
+                        devices[device_id] = {
+                            CONF_NAME: bulb["ipaddr"],
+                            CONF_HOST: bulb["ipaddr"],
+                        }
+
+            return self.async_create_entry(
+                title="FluxLED/MagicHome",
+                data={
+                    CONF_AUTOMATIC_ADD: user_input[CONF_AUTOMATIC_ADD],
+                    CONF_EFFECT_SPEED: DEFAULT_EFFECT_SPEED,
+                    CONF_DEVICES: devices,
+                },
+            )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_AUTOMATIC_ADD, default=True): bool}
```

Comment:
I think this should just be an option in the options flow. We should default to True without showing that option in the user step in the main flow. This form could just be a confirm dialog.

Reply:
I really like having the option on setup. Personally I don't want them to automatically get added / scan on install because I would prefer to set them up individually. It appears from (small) feedback I've received there are several like that. I'm not sure the benefit of taking away this as an option.

Reply:
Ok.